e2e/nextjs/go-service/main.go

```diff
@@ -0,0 +1,131 @@
+// main.go
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/highlight/highlight/sdk/highlight-go"
+
+	highlightChi "github.com/highlight/highlight/sdk/highlight-go/middleware/chi"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type MyLogger struct{}
+
+func (l MyLogger) Error(v ...interface{}) {
+	fmt.Println(v...)
+}
+
+func (l MyLogger) Errorf(format string, v ...interface{}) {
+	fmt.Println(fmt.Sprintf(format, v...))
+}
+
+var logger highlight.Logger = MyLogger{}
+
+func main() {
+	highlight.SetDebugMode(logger)
+	highlight.SetOTLPEndpoint("http://localhost:4318")
+	highlight.Start(
+		highlight.WithServiceName("my-go-service"),
+		highlight.WithProjectID("1"),
+
+		highlight.WithSamplingRateMap(map[trace.SpanKind]float64{
```

Comment:
Not blocking, but this should no longer be needed as the issue was hopefully fixed in https://github.com/highlight/highlight/pull/8086